pkg/receive/limits/limiter.go

```diff
@@ -0,0 +1,191 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package limits
```

Comment:
Any reason to separate this into a package? I know the PR description says it's being done in here, but I'm wondering about the reason since we do not have other sub-packages in `receive`.

Reply:
Just for organizational purposes. It also starts facilitating little by little future code reusability for some of these limits (i.e. per request limits).

Reply:
I still don't see that as a strong reason given that's how all other parts of `receive` are organized, but not a blocker.

Reply:
FYI I got rid of this package in the recent commits. It also make conflict resolutions much easier.